cmake/LyAutoGen.cmake

```diff
@@ -17,15 +17,15 @@ function(ly_add_autogen)
 
     set(options)
     set(oneValueArgs NAME)
-    set(multiValueArgs INCLUDE_DIRECTORIES AUTOGEN_RULES ALLFILES)
+    set(multiValueArgs INCLUDE_DIRECTORIES AUTOGEN_RULES ALLFILES FILES_CMAKE)
     cmake_parse_arguments(ly_add_autogen "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
     if(ly_add_autogen_AUTOGEN_RULES)
         set(AZCG_INPUTFILES ${ly_add_autogen_ALLFILES})
         list(FILTER AZCG_INPUTFILES INCLUDE REGEX ".*\.(xml|json|jinja)$")
         target_include_directories(${ly_add_autogen_NAME} PUBLIC "${CMAKE_CURRENT_BINARY_DIR}/Azcg/Generated")
         execute_process(
-            COMMAND ${LY_PYTHON_CMD} "${LY_ROOT_FOLDER}/cmake/AzAutoGen.py" "${CMAKE_BINARY_DIR}/Azcg/TemplateCache/" "${CMAKE_CURRENT_BINARY_DIR}/Azcg/Generated/" "${CMAKE_CURRENT_SOURCE_DIR}" "${AZCG_INPUTFILES}" "${ly_add_autogen_AUTOGEN_RULES}" "-n"
+            COMMAND ${LY_PYTHON_CMD} "${LY_ROOT_FOLDER}/cmake/AzAutoGen.py" "${CMAKE_BINARY_DIR}/Azcg/TemplateCache/" "${CMAKE_CURRENT_BINARY_DIR}/Azcg/Generated/" "${CMAKE_CURRENT_SOURCE_DIR}" "${ly_add_autogen_FILES_CMAKE}" "${ly_add_autogen_AUTOGEN_RULES}" "-n"
```

Comment:
I think a better solution instead of trying to parse a cmake file, is to use the FILTER `AZCG_INPUTFILES` list and use the [file(CONFIGURE)](https://cmake.org/cmake/help/latest/command/file.html#configure) command to generate a simple line semicolon delimited text file, that is then supplied to AzAutoGen.py

```
# Template for semicolon delimited list of AZ autogen input files
set(autogen_input_files_template [[
@AZCG_INPUTFILES@
]])
# Writes the list of AzAutoGen input files into a temporary ${ly_add_autogen_NAME}_input_files.txt file
set(input_files_path "${CMAKE_CURRENT_BINARY_DIR}/Azcg/Temp/${ly_add_autogen_NAME}_input_files.txt"
file(CONFIGURE OUTPUT "${input_files_path}" "${autogen_input_files_template}" @ONLY)

execute_process(
    COMMAND ${LY_PYTHON_CMD} "${LY_ROOT_FOLDER}/cmake/AzAutoGen.py" "${CMAKE_BINARY_DIR}/Azcg/TemplateCache/" "${CMAKE_CURRENT_BINARY_DIR}/Azcg/Generated/" "${CMAKE_CURRENT_SOURCE_DIR}" ""${input_files_path}"" "${ly_add_autogen_AUTOGEN_RULES}" "-n"
```

Reply:
done